TeamCode/src/main/java/org/firstinspires/ftc/teamcode/MecanumDrive.java

```diff
@@ -199,16 +213,18 @@ public Twist2dDual<Time> update() {
 
             lastHeading = heading;
 
-            return new Twist2dDual<>(
+            delta = new Twist2dDual<>(
```

Comment:
Making the `delta` is kind of unnecessary here because it's immediately split. You should be able to pass `Twist2d(twist.line.value(), headingDelta)` directly to `pose.plus()` and then return `twist.velocity().value()`.